app/Http/Controllers/SponsorLoginController.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Controllers;
+
+use App\Models\Sponsor;
+use App\Models\SponsorDomain;
+use App\Models\SponsorUser;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+
+class SponsorLoginController
+{
+    public function showLoginForm()
+    {
+        return view('sponsors.login');
+    }
+
+    public function validateEmail(Request $request): JsonResponse
+    {
+        // Validate input request using Laravel's in-built validator
+        $request->validate([
+            'email' => [
+                'required',
+                'string',
+                'email:rfc,strict,dns,spoof',
+                'max:255',
+            ],
+        ]);
+
+        // Read value - cast to string since validation guarantees it
+        $email = (string) $request->input('email');
+
+        // Check if domain is valid and sponsor is active; if not, return JSON error
```

Comment:
I would gently suggest removing comments that directly restate the logic in the code; this is above the amount of comments generally found in this codebase